{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Geneious Protocol\n",
        "\n",
        "## Overview\n",
        "\n",
        "This protocol provides a standardized method for building consensus sequences from raw sequence files using Geneious software. It includes steps for uploading raw files, assembling contigs, and conducting quality control during sequence analysis. The method is adapted from the [SI Barcode Network Informatics Documentation](https://sibarcodenetwork.readthedocs.io/en/latest/sequence_qc.html) and Geneious tutorials.\n",
        "\n",
        "## Background\n",
        "\n",
        "Standardizing sequence analysis methods ensures repeatability in research and transparency in methodology, which are essential for future studies and peer evaluations.\n",
        "\n",
        "---\n",
        "\n",
        "## Methodology\n",
        "\n",
        "### Step 1: Folder Organization\n",
        "\n",
        "Organize your workspace in Geneious. Suggested folder structure for **each gene marker**:\n",
        "\n",
        "-   **01_Raw Sequences**\n",
        "\n",
        "-   **02_Trimmed Sequences**\n",
        "\n",
        "-   **03_Assembled Sequences**\n",
        "\n",
        "-   **04_Final Sequences for Alignment**\n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 2: Add Primers\n",
        "\n",
        "Add primer sequences to Geneious for use during trimming below.\n",
        "\n",
        "1. Create new folder called \"Primers\"\n",
        "2. Go to **Sequence - New Sequence**.\n",
        "3. Copy the primer sequence into the respective field.\n",
        "4. In the \"Name\" field, enter the primer name (e.g., CYTB151F).\n",
        "5. In the \"Description\" field, enter \"primer\" plus the direction (e.g., primer forward).\n",
        "6. Change \"Type\" to \"Primer\".\n",
        "7. Press OK.\n",
        "8. Repeat Steps 2-7 for each primer.\n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 3: Upload and Prepare Raw Files\n",
        "\n",
        "1. Upload raw sequence files into the **01_Raw Sequences** folder.  \n",
        "2. Rename files to include sample ID, extraction ID, location, morph, and primer.  \n",
        "   Example: `BFHL-4505_BC111_WA_RB_CYTB151F`.  \n",
        "3. Set read directions for all sequences.  \n",
        "\n",
        "---\n",
        "\n",
        "### Step 4: Trim Poor-Quality Bases\n",
        "\n",
        "1. Copy sequences to the **02_Trimmed Sequences** folder.  \n",
        "2. Use the **Annotate & Predict - Trim Ends...** function:  \n",
        "   - Check **Annotate new trimmed regions**\n",
        "   - Check **Trim primers** and select your respective forward and reverse primers.\n",
        "   - Set Allow Mismatches to 2.\n",
        "   - Set Minimum Match Length to 10.\n",
        "   - Set **Error Probability Limit** to 0.05.\n",
        "   - Select **Trim 5’ End** and **Trim 3’ End**. \n",
        "   - Leave everything else unselected.\n",
        "3. Review the trimming annotations and manually adjust if necessary. For good sequences or specific regions, create custom trim areas if needed.  \n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 5: Sequence Renaming\n",
        "\n",
        "1. Copy sequences into a new folder with the same name followed by **“_renamed”**.\n",
        "2. Use **Edit - Batch rename...** to rename sequences for contig assembly. This will make the forward and reverse sequence names identical so that Geneious knows which ones to combine. \n",
        "    - First, remove the primer name and file extension from the end of the sequence name.\n",
        "        - Check **Remove** and enter the number of characters to remove from the end of the sequence name so that the primer name and file extension are removed.\n",
        "        - Example: `BFHL-4505_BC111_WA_RB_CYTB151F.ab1` becomes `BFHL-4505_BC111_WA_RB`\n",
        "    - Second, remove the extraction name from the middle of the sequence name \n",
        "        - Under \"Advanced\", check **Regex** and enter \"(\\w{5})_\" in the **Replace parts matching:** field.\n",
        "        - Enter nothing in the **With:** field.\n",
        "        - Example: `BFHL-4505_BC111_WA_RB` becomes `BFHL-4505_WA_RB`\n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 6: Assemble Contigs\n",
        "\n",
        "1. Assemble sequences into contigs based on read names (i.e., based on the first space). Use the following parameters:\n",
        "    - **Assembler:** Geneious\n",
        "    - **Sensitivity:** Highest Sensitivity/Slow\n",
        "    - Check **Use existing trim regions**.\n",
        "    - **Assembly Name:** {Reads Name}\n",
        "    - Check **Save assembly report**.\n",
        "    - If desired, check **Save in sub-folder**.\n",
        "    - Check **Save contigs**.\n",
        "    - Set **Don't merge variants with coverage over approximately** to 6\n",
        "    - Check **Merge homopolymer variants**\n",
        "2. Move assembly report and assemblies to the **03_Assembled** folder.  \n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 7: Quality Control\n",
        "1. Manually review contigs in assembly report. Successful assemblies are those which have 75% HQ bases and above.\n",
        "    - Move assembled contigs with $\\geq$ 75% HQ bases to **Contig_assembled_highquality** subfolder.\n",
        "    - Move assembled contigs which have $\\leq$75% HQ bases to **Contig_assembled_lowquality** subfolder\n",
        "    - Copy failed reads to **Contig_not_assembled** subfolder.\n",
        "    - Move failed reads which have $\\geq$75% HQ bases to **Contig_not_assembled_goodread** subfolder\n",
        "    \n",
        "2. Address base call disagreements and gaps. For each sequence, use CTRL+D to look at each base call disagreement and gap region. Make sure that you agree with the Geneious calls. If not, edit accordingly.  \n",
        "   - Check if gap is due to dye blob or error (i.e., a repeated base). This is likely caused by a chromatogram reading error where there a longer signal than usual was picked up due to dye.\n",
        "   - Check if gap is due to F and R alignment error. Shifts in the reading frame usually means a nucleotide or gap was added to one of the traces incorrectly. This type of error can usually be corrected by inspecting the properly aligned section of the sequence to determine the location of the mistake. \n",
        "   - If neither of the above, fill gap with the best-guess bp based on the chromatogram. Fill with heterozygous base, if unsure.\n",
        "\n",
        "3. Align consensus sequences from contigs (ignore single reads for now). Analyzing the sequences’ alignment will inform of any further end trimming needed if the Geneious Prime Assembler neglected to remove primers or poor quality end regions.\n",
        "    - Go to **Align/Assemble - Multiple Align...**.\n",
        "    - Select **Consensus Align** in the top portion of the window.\n",
        "    - Set **Threshold** to 65% in the **Consensus Options** menu.\n",
        "    - In the dropdown **Sequence alignment options** menu, select **MUSCLE alignment**\n",
        "    - Click “OK”.\n",
        "    - Rename alignment as \"Contig alignment\".\n",
        "    - First, check for problematic sequences that are introducing gaps due to incorrect editing during Step 2. If any are present, right click on the sequence and select \"Go to referenced sequence\", and re-edit accordingly.\n",
        "    - Check alignment ends for untrimmed primer regions or poor quality regions that need to be removed. Delete these areas and hit Save.\n",
        "    - Make sure there are no stop codons in the sequences.\n",
        "        - In the **Display** tab on the right, check **Translation** on \"All Sequences\".\n",
        "        - Set **Genetic Code:** to \"Invertebrate Mitochondrial\"\n",
        "        - Set **Relative to:** to \"Alignment\"\n",
        "        - Try each reading frame and make sure there is one where there are no stop codons in any of your sequences. If not, inspect and edit your sequences like above until this is fixed.\n",
        "        \n",
        "4. Exit the alignment and generate consensus sequences from the final contig set. \n",
        "    - Select all sequences within the alignment.\n",
        "    - Right click and select **Extract Regions...**.\n",
        "    - Check **Extract region as list of sequences** and press OK.\n",
        "    - Move the resulting list to the **Consensus_preliminary** subfolder. Right click on the list and select **Extract Sequences from List...** to extract sequences into the folder. Delete the \"Consensus\" file and list file since you do not need these.\n",
        "\n",
        "5. Copy high quality and chosen intermediate quality single reads to the same **Consensus_preliminary** folder. For each single read, go to **Trim Ends..** and check **Remove existing trimmed regions from sequences**. Hit OK.\n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 8: Final Quality Check\n",
        "\n",
        "1.  Align final consensus sequences and good/intermediate single reads, and make sure everything looks OK.\n",
        "    - Go to **Align/Assemble - Multiple Align...**.\n",
        "    - Select **MUSCLE Alignment** in the top portion of the window.\n",
        "    - Click “OK”.\n",
        "    - Rename alignment file as \"Consensus alignment\".\n",
        "    - Check alignment for gaps, untrimmed ends, and reading frame stop codons. You might have to **reverse complement** single reads and/or trim their ends more to match up with the contig consensus sequences.\n",
        "    \n",
        "2. BLAST sequences to ensure they belong to the target group. If doing this in Geneious, it is recommended to only BLAST small batches of 15 or less sequences at a time since it can become time consuming.\n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### Step 11: Export Final Consensus Sequences\n",
        "\n",
        "1. Export consensus sequences into a single FASTA file.\n",
        "2. Proceed to further alignment and phylogenetic analyses using tools like MEGA.  \n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "This structured approach ensures a robust and repeatable pipeline for sequence analysis using Geneious software.\n"
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.13.1"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 4
}
